security/pkg/util/certutil.go

```diff
@@ -27,6 +27,8 @@ import (
 type CertUtil interface {
 	// GetWaitTime returns the waiting time before renewing the certificate.
 	GetWaitTime([]byte, time.Time, time.Duration) (time.Duration, error)
+	// GetMinimumTTL returns the minimum duration of either the expire date of the intermediate certificate or the workload TTL
+	GetMinimumTTL(certBytes []byte, now time.Time, workloadTTL time.Duration) time.Duration
```

Comment:
Is it possible to change this interface as 

```go
GetWaitTimeForCert(certChain, now time.Time, gracePeriod time.Duration)
```

The check logic of the intermediate CA TTL must be respected can be done in the implementation of this function itself.

And `GetWaittime` can be migrated as well.

Reply:
And from the root to the leaf, all TTL should be examined, not just one particular intermediate CA one. Am I correct?